.github/workflows/publish.yml

```diff
@@ -0,0 +1,87 @@
+name: Publish helm chart and docker images
+
+# Trigger the workflow on tags or commits to the main or master branch.
+on:
+  push:
+    branches: [main, master]
+    tags:
+
+defaults:
+  run:
+    # Declare bash be used by default in this workflow's "run" steps.
+    #
+    # NOTE: bash will by default run with:
+    #   --noprofile: Ignore ~/.profile etc.
+    #   --norc:      Ignore ~/.bashrc etc.
+    #   -e:          Exit directly on errors
+    #   -o pipefail: Don't mask errors from a command piped into another command
+    shell: bash
+
+jobs:
+  # Builds and pushes docker images to DockerHub, packages the Helm chart and
+  # pushes it to jupyterhub/helm-chart@gh-pages where index.yaml represents the
+  # JupyterHub organization Helm chart repository.
+  #
+  # ref: https://github.com/jupyterhub/helm-chart
+  # ref: https://hub.docker.com/orgs/jupyterhub
+  #
+  Publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # chartpress requires the full history
+          fetch-depth: 0
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Install chart publishing dependencies (chartpress, helm)
+        run: |
+          . ./ci/common
+          setup_helm v3.4.0
```

Comment:
Does someone know a way to share/get this version from the action defined in `test.yml`? Would be nice but not a big deal I think

Reply:
I googled a bit and didn't find a solution. I think workflows run independently of each other, so either you need to explicitly import one from the other somehow or declare something outside both that is read by both, like a github secret variable. I think those approaches are introducing more complexity that makes it harder to both maintain and understand to motivate not breaking the DRY principle.

We can use the stable version in both as an alternative, by not providing an explicit version. The installer script to get helm gives you the latest version with a delay of a day or so i think. At least I got 3.4.0 when 3.4.1 had been out for some hours already.

Reply:
For me it is ok to repeat it here as it shouldn't matter much which version is used here. Especially as there is no easy way to share. And probably easy to find if it does lead to breakage.